Cython/StringIOTree.py

```diff
@@ -107,3 +107,32 @@ def insertion_point(self):
     def allmarkers(self):
         children = self.prepended_children
         return [m for c in children for m in c.allmarkers()] + self.markers
+
+    # Print the result of allmarkers in a nice human-readable form. Use it only for debugging.
+    # Ptints e.g.
+    # cython line 2 maps to 3299-3343
+    # cython line 4 maps to 2236-2245  2306  3188-3201
+    # ...
+    # Note: In the example above, 3343 maps to line 2, 3344 does not.
+    def print_hr_allmarkers(self):
+        from collections import defaultdict
+        markers = self.allmarkers()
+        d = defaultdict(list)
+        for c_lineno, cython_lineno in enumerate(markers):
+            if cython_lineno > 0:
+                d[cython_lineno].append(c_lineno + 1)
+        if len(d) == 0:
+            print("allmarkers is empty")
+        for cython_lineno, c_linenos in sorted(d.items()):
+            sys.stdout.write("cython line", cython_lineno, "maps to ")
```

Comment:
This doesn't work (and is obviously untested). `write()` only takes a single argument. You have to build the output string first. In fact, I'd build a list of line numbers first and then output it in the end. That avoids timing issues with potentially concurrent output.

Reply:
Fixed that in the newest commit